src/clj/fluree/db/query/exec/where.cljc

```diff
@@ -415,14 +412,14 @@
 
 (defmethod match-pattern :tuple
   [ds fuel-tracker solution pattern filters error-ch]
-  (let [active-graph (cond-> ds
-                       (dataset/dataset? ds) dataset/active)]
+  (if-let [active-graph (dataset/active ds)]
     (if (sequential? active-graph)
       (->> active-graph
            (map (fn [graph]
                   (match-tuple graph fuel-tracker solution pattern filters error-ch)))
            async/merge)
-      (match-tuple active-graph fuel-tracker solution pattern filters error-ch))))
+      (match-tuple active-graph fuel-tracker solution pattern filters error-ch))
+    (go)))
```

Comment:
Oh, is it just meant to return a channel with nothing in it, because the other `if` branches return a channel? That makes sense, though maybe you could help me understand when we wouldn't have an active graph.

Reply:
I just changed it to reuse the same closed channel so as not to create any unnecessary go blocks. not a big deal if we did, but there's no need to in this situation. this is just the equivalent of `nil` in the async execution environment. 

We don't have an active graph in the case where a query specifies a `from-named` clause but no `from` clause, and the particular where clause pattern does not specify a named graph. We used to throw an inscrutable error, but it actually just means there's no data to query. We could throw another error during validation to catch this situation, and I went back and forth over whether we should do that, but I think that's outside the scope of this work and doesn't match with how we treat other situations like this: if there's no data to query, we return an empty set.